tools/storm/scripts/acr.go

```diff
@@ -0,0 +1,186 @@
+}
+
+func generateTagBase(buildId string, config string, deploymentEnvironment string) string {
+	return fmt.Sprintf("v%s.%s.%s", buildId, config, deploymentEnvironment)
+}
+
+func loginToACR(acrName string) error {
+	logrus.Infof("Logging in to ACR: %s", acrName)
+	cmd := exec.Command("az", "acr", "login", "-n", acrName)
+	return cmd.Run()
+}
+
+// Define AcrPushScript
+type AcrPushScript struct {
+	Config                string   `required:"" help:"Trident configuration (e.g., 'extensions')" type:"string"`
+	DeploymentEnvironment string   `required:"" help:"Deployment environment (virtualMachine or bareMetal)" type:"string"`
+	AcrName               string   `required:"" help:"Azure Container Registry name" type:"string"`
+	RepoName              string   `required:"" help:"Repository name in ACR" type:"string"`
+	BuildId               string   `required:"" help:"Build ID" type:"string"`
+	FilePaths             []string `required:"" help:"Array of file paths to push to ACR"`
+}
+
+func (s *AcrPushScript) Run() error {
+	// Login to ACR
+	err := loginToACR(s.AcrName)
+	if err != nil {
+		return fmt.Errorf("failed to login to ACR: %w", err)
+	}
+
+	// Push all specified files
+	tagBase := generateTagBase(s.BuildId, s.Config, s.DeploymentEnvironment)
+	err = s.pushFiles(tagBase)
+	if err != nil {
+		return fmt.Errorf("failed to push files: %w", err)
+	}
+
+	// Set output variable by writing to stdout
+	fmt.Printf("##vso[task.setvariable variable=TAG_BASE]%s\n", tagBase)
+	logrus.Infof("TAG_BASE set to: %s", tagBase)
+
+	return nil
+}
+
+func (s *AcrPushScript) pushFiles(tagBase string) error {
+	for i, filePath := range s.FilePaths {
+		// Check if file exists
+		if _, err := os.Stat(filePath); os.IsNotExist(err) {
+			return fmt.Errorf("file %s does not exist: %w", filePath, err)
+		}
+
+		// Create tag with index
+		tag := fmt.Sprintf("%s.%d", tagBase, i+1)
+
+		// Push the file
+		err := s.pushImage(filePath, tag)
+		if err != nil {
+			return fmt.Errorf("failed to push file %s: %w", filePath, err)
+		}
+
+		// Verify the push
+		err = s.verifyImage(s.RepoName, tag)
+		if err != nil {
+			return fmt.Errorf("failed to verify %s:%s: %w", s.RepoName, tag, err)
+		}
+	}
+
+	return nil
+}
+
+func (s *AcrPushScript) pushImage(filePath, tag string) error {
+	registryURL := fmt.Sprintf("%s.azurecr.io", s.AcrName)
+	fullImageName := fmt.Sprintf("%s/%s:%s", registryURL, s.RepoName, tag)
+
+	logrus.Infof("Pushing %s with tag %s to %s", filePath, tag, registryURL)
+
+	// Get the directory and filename from the full path
+	dir := filepath.Dir(filePath)
+	fileName := filepath.Base(filePath)
+
+	// Use ORAS to push the image
+	cmd := exec.Command("oras", "push", fullImageName, fileName)
+	cmd.Dir = dir
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		logrus.WithField("output", string(output)).Errorf("Failed to push %s with oras", fullImageName)
+		return err
+	}
+
+	// Sleep to allow registry to process
+	time.Sleep(3 * time.Second)
```

Comment:
instead of doing an arbitrary wait, implement a retry look on verify image with a modest back-off :)
not performance-critical code, but generally a good practice to follow